mono/metadata/marshal.c

```diff
@@ -9365,6 +9365,7 @@ mono_marshal_get_synchronized_wrapper (MonoMethod *method)
 #endif
 
 	if (method->klass->valuetype && !(method->flags & MONO_METHOD_ATTR_STATIC)) {
+		/* FIXME Is this really the best way to signal an error here?  Isn't this called much later after class setup? -AK */
 		mono_class_set_type_load_failure (method->klass, "");
```

Comment:
Didn't change this one, but I want to revisit it later.  It seems like `mono_marshal_get_synchronized_wrapper` should just use our normal `MonoError` mechanism to signal and error here.